.rfc/RFC0000-rfc-process.md

```diff
@@ -0,0 +1,156 @@
+---
+RFC:          RFC0000
+Author:       Mikey Lombardi
+Status:       Draft
+SupercededBy: N/A
+Version:      1.0
+Area:         Process
+---
+
+# DSC RFC Process and Guidelines
+
+A DSC RFC (Request for Comments) is a publication to propose design changes and improvements to
+DSC. This provides the community an opportunity to provide feedback before code is written where it
+becomes harder to change at the risk of compatibility.
+
+This process was adapted from the PowerShell RFC process, which itself was adapted from the Chef
+RFC process and the DMTF.org process.
```

Comment:
Since we don't want/need to have all the docs from the PowerShell repo, we can probably just link to relavent docs in the PS repo as applicable here.  For example, I would still prefer community members to start with an issue/discussion instead of spending time on an RFC.  This is somewhat covered below in the roles, but I'd rather put it up front so we don't suddenly get a bunch of RFC PRs.